doc/ACL/SAI-Proposal-Exact-Match.md

```diff
@@ -0,0 +1,123 @@
+```
+typedef enum _sai_acl_match_type_t
+{
+    /** ternary matching */
+    SAI_ACL_MATCH_TYPE_TERNARY,
+
+    /** exact matching */
+    SAI_ACL_MATCH_TYPE_EXACT,
+
+} sai_acl_match_t;
+```
+
+3. Add capability enhancments to query SDK to determine the match types that an ACL stage can support.
+```
+/**
+ * @brief Attribute data for SAI_ACL_TABLE_ATTR_SUPPORTED_MATCH_TYPE
+ */
+typedef enum _sai_acl_supported_match_type_t
+{
+    /** Ternary */
+    SAI_ACL_SUPPORTED_MATCH_TYPE_TERNARY,
+
+    /** Exact */
+    SAI_ACL_SUPPORTED_MATCH_TYPE_EXACT,
+
+    /** Ternary and Exact */
+    SAI_ACL_SUPPORTED_MATCH_TYPE_TERNARY_AND_EXACT,
+
+} sai_acl_supported_match_type_t;
+
+typedef struct _sai_acl_capability_t
+{
+    :
+    /**
+     * @brief Supported match type
+     *
+     * Indicates the match type supported by ACL stage
+     */
+    sai_acl_supported_match_type_t supported_match_type;
+
+} sai_acl_capability_t;
+```
+
+4. Add new attributes to ACL table to specify valid bits for keys.  If a match field does not have valid bits specified, all bits in the field are valid. For tables implemented using Exact Match, there is no further key masking supported in table entries. The mask for such entries needs to be set to all 1s.
+```
+    /**
+     * @brief Src IPv6 Valid bits
+     *
+     * @type sai_acl_field_data_mask_t sai_ip6_t
+     * @flags CREATE_ONLY
+     * @default ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
+     * @validonly SAI_ACL_TABLE_ATTR_FIELD_SRC_IPV6 == true
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_VALID_BITS_SRC_IPV6 = SAI_ACL_TABLE_ATTR_FIELD_VALID_BITS_START,
+```
+
+# Example Call Sequence #
+The following new attributes will be set for creating an exact match ACL entry with a dest ipv6 prefix.
+1. Acl Table group  create - create_acl_table_group with attributes.
+```
+==========================================================================
+|  Attribute Name                     |          Attribute Value          |
+===========================================================================
+| SAI_ACL_TABLE_GROUP_ATTR_ACL_STAGE  | SAI_ACL_STAGE_POST_INGRESS        |
+==========================================================================
+```
+2. Acl Table create - create_acl_table with following attributes.
+```
+============================================================================================
+|  Attribute Name                              |                   Attribute Value          |
+============================================================================================
+| SAI_ACL_TABLE_ATTR_ACL_STAGE                 | SAI_ACL_STAGE_POST_INGRESS                 |
+| SAI_ACL_TABLE_ATTR_ACL_MATCH_TYPE            | SAI_ACL_MATCH_TYPE_EXACT                   |
+| SAI_ACL_TABLE_ATTR_FIELD_DST_IPV6            |  true                                      |
+| SAI_ACL_TABLE_ATTR_FIELD_DATA_MASK_DST_IPV6  |  sai_acl_field_data_mask_t with valid bits |
```

Comment:
This example seems to suggest user can pass different combination of valid bits.

Reply:
it is based on capability - https://github.com/opencomputeproject/SAI/pull/1730/files#diff-444febffce442e4277a1ced22d1ab94df045dfaa10a1b671ee35a7e5c2d9f683R836